src/Sylius/Behat/Page/Admin/Product/IndexPage.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Page\Admin\Product;
+
+use Behat\Mink\Session;
+use Sylius\Behat\Page\SymfonyPage;
+use Sylius\Behat\TableManipulatorInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class IndexPage extends SymfonyPage implements IndexPageInterface
+{
+    /**
+     * @var TableManipulatorInterface
+     */
+    private $tableManipulator;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param TableManipulatorInterface $tableManipulator
+     */
+    public function __construct(
+        Session $session,
+        array $parameters,
+        RouterInterface $router,
+        TableManipulatorInterface $tableManipulator
+    ) {
+        parent::__construct($session, $parameters, $router);
+
+        $this->tableManipulator = $tableManipulator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isThereProduct(ProductInterface $product)
+    {
+        if (!$table = $this->getDocument()->find('css', 'table')) {
+            return false;
```

Comment:
I thought we display empty table, so when there will be no table then it means something is wrong with the page and we should throw exception. Taking into account that no results = no table current implementation works fine :)